rpisense-core.c

```diff
@@ -20,37 +20,24 @@
 #include <linux/init.h>
 #include <linux/i2c.h>
 #include <linux/platform_device.h>
-#include "core.h"
+#include "rpisense.h"
 #include <linux/slab.h>
 
-static void rpisense_client_dev_register(struct rpisense *rpisense,
-					 const char *name,
-					 struct platform_device **pdev)
-{
-	int ret;
+#define RPISENSE_FB			0x00
+#define RPISENSE_WAI			0xF0
+#define RPISENSE_VER			0xF1
+#define RPISENSE_KEYS			0xF2
+#define RPISENSE_EE_WP			0xF3
 
-	*pdev = platform_device_alloc(name, -1);
-	if (*pdev == NULL) {
-		dev_err(rpisense->dev, "Failed to allocate %s\n", name);
-		return;
-	}
+#define RPISENSE_ID			's'
 
-	(*pdev)->dev.parent = rpisense->dev;
-	platform_set_drvdata(*pdev, rpisense);
-	ret = platform_device_add(*pdev);
-	if (ret != 0) {
-		dev_err(rpisense->dev, "Failed to register %s: %d\n",
-			name, ret);
-		platform_device_put(*pdev);
-		*pdev = NULL;
-	}
-}
+static struct platform_device *
```

Comment:
kind of stands out to have this lone forward decl when we could just move the function to this location of the code

Reply:
I wanted probe and remove at the top of each file. When I rewrote frame buffer I also had to add a single forward decl. to make that possible. Obviously this is a matter of taste, but I think I prefer probe and remove at top as opposed to whatever other random crap. makes it easier to follow the flow of code